BUG: Allow multiple DICOM plugins for export with same mean confidence

Sunderlandkyl opened the pull request:
Previously only one plugin was stored within the QMap for a given mean confidence value. This commit changes the QMap to a QMultiMap so that multiple plugins with the same confidence can be displayed.

See discussion here: https://discourse.slicer.org/t/exporting-segmentation-as-dcm/28168

jcfr commented:
@Sunderlandkyl Could you consolidate this with commit `COMP: Fix -Wdeprecated-declarations in qSlicerDICOMExportDialog.cxx` associated with https://github.com/Slicer/Slicer/pull/6864 ?

pieper reviewed (dismissed):
LGTM 👍

Sunderlandkyl commented:
@jcfr Sure, I brought over the changes from https://github.com/Slicer/Slicer/pull/6864, although there was a bug causing duplicate entries in the list. I fixed that issue and added back the QList to exportablesByConfidence so that exportables remained grouped by plugin.

To clarify, the previously intended behavior seems to be to create only one entry in the list for each plugin, with the number of exportables displayed as "(# series)". Is this still the intended behavior, or do we want an entry for each exportable?:
![image](https://user-images.githubusercontent.com/9222709/223565619-ab472207-b33f-407b-aa97-6440a03aeabd.png)

jcfr commented:
Thanks for consolidating.

> added back the QList to exportablesByConfidence so that exportables remained grouped by plugin

Makes sense. This was an oversight on my side. The behavior you described is the correct one.

jcfr commented on Modules/Scripted/DICOMLib/Widgets/qSlicerDICOMExportDialog.cxx:
```diff
@@ -289,24 +289,16 @@ void qSlicerDICOMExportDialog::examineSelectedItem()
   // Map the grouped exportables by confidence values so that the highest confidence is on top
-  QMap<double,QList<qSlicerDICOMExportable*> > exportablesByConfidence;
-  foreach(QList<qSlicerDICOMExportable*> exportablesForPlugin, exportablesByPlugin)
+  QMultiMap<double,QList<qSlicerDICOMExportable*> > exportablesByConfidence;
```
When using `QMultiMap`, the values associated with a given key are managed using a list. For this reason, I think using `QMultiMap<double,qSlicerDICOMExportable*> exportablesByConfidence;` should be enough ?
